engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/number/utils/BigIntegerOps.java

```diff
@@ -235,7 +235,12 @@ public static boolean fitsInLong(BigInteger bigInteger) {
   }
 
   public static boolean fitsInLong(double decimal) {
-    return decimal <= Long.MAX_VALUE && decimal >= Long.MIN_VALUE;
+    var nulaMinus = Double.doubleToRawLongBits(-0d);
+    if (nulaMinus == Double.doubleToRawLongBits(decimal)) {
+      return false;
+    }
```

Comment:
Does that mean that `-0.0` will no longer be equal to `0`?

I'd expect this to be true:
```
> 1/Number.positive_infinity
>>> 0.0
> -1/Number.positive_infinity
>>> -0.0
> (1/Number.positive_infinity) == (-1/Number.positive_infinity)
>>> true
> (-1/Number.positive_infinity) == 0
>>> true
```
(that's how it works currently)

I think we should add a test case to make sure it works as expected. In particular, we need to ensure we don't break transitivity of `==`: `0 == 0.0 == -0.0`

Reply:
I guess this will work because the `0` literal is `long`. But what if I create a `0` that is a `BigInteger`? It definitely must be consistent.

Reply:
I had to return this change back and rather do d69f62d where needed.